tests/backwards_compatibility_tests/compatibility_test_runner.py

```diff
@@ -0,0 +1,603 @@
+            subprocess.run(["docker", "pull", image_identifier], check=True)
+            return image_identifier
+        elif source == "ECR":
+            return pull_remote_image_from_ecr(image_digest=image_identifier)
+    except subprocess.CalledProcessError as e:
+        raise Exception(f"Failed to pull Docker image {image_identifier}: {e}")
+
+
+def start_marqo_from_version_container(version: str, from_version_volume, from_version_image: Optional[str] = None,
+                                       env_vars: Optional[list] = None):
+    """
+    Start a Marqo container after pulling the required image from docker and creating a volume.
+    The volume is mounted to a specific point such that it can be later used to transfer state to a different version of Marqo.
+
+    Args:
+        version (str): The version of the Marqo container to start.
+        from_version_volume: The volume to use for the container.
+        from_version_image (Optional[str]): The specific image to use for the container. Defaults to None.
+        env_vars (Optional[list]): A list of environment variables to set in the container. Defaults to None.
+    """
+
+    source = "docker" #The source for from_version image would always be docker because it's supposed to be an already released docker image
+
+    logger.debug(f"Starting Marqo container with version {version}, from_version_image {from_version_image}, from_version_volume {from_version_image}, source {source}")
+    from_version_image = from_version_image or f"marqoai/marqo:{version}"
+    container_name = f"marqo-{version}"
+
+    logger.debug(f"Using image: {from_version_image} with container name: {container_name}")
+
+    # Pull the image before starting the container
+    pull_marqo_image(from_version_image, source)
+
+    # Stop and remove the container if it exists
+    try:
+        subprocess.run(["docker", "rm", "-f", container_name], check=True)
+    except Exception as e:
+        logger.debug(f"Container {container_name} not found, skipping removal.")
+
+    # Prepare the docker run command
+    cmd = [
+        "docker", "run", "-d",
+        "--name", container_name,
+        "-it", "-p", "8882:8882",
+        "-e", "MARQO_ENABLE_BATCH_APIS=TRUE",
+        "-e", "MARQO_MAX_CPU_MODEL_MEMORY=1.6"
+    ]
+
+    # Append environment variables passed via the method
+    if env_vars:
+        for var in env_vars:
+            cmd.extend(["-e", var])
+
+    # Handle version-specific volume mounting
+
+    # Mounting volumes for Marqo >= 2.9
+    # Use the provided volume for state transfer
+    from_version_volume = create_volume_for_marqo_version(version, from_version_volume)
+    logger.debug(f"from_version volume = {from_version_volume}")
+    if version >= marqo_transfer_state_version:
+        # setting volume to be mounted at /opt/vespa/var because starting from 2.9, the state is stored in /opt/vespa/var
+        cmd.extend(["-v", f"{from_version_volume}:/opt/vespa/var"])
+    else:
```

Comment:
will we ever need to test marqo prior to 2.9?

Reply:
Not sure, but for the sake of completeness I've added the logic to do so